### ISSUE TITLE
Refactor Upscale*DLayer

### ISSUE THREAD

**Issue by f0k** [Lasagne/Lasagne#737]
Upscale1DLayer, Upscale2DLayer, Upscale3DLayer share a lot of code (and tests). They could be refactored to be derived from a common UpscaleLayer that supports any number of spatial dimensions, or maybe even any set of axes (so it could be used to repeat channels or timesteps).

**Comment by andrewbwogi** [comment]
Hi, I'm new to open source development and would like to contribute. Can I work on this issue?

**Comment by f0k** [comment]
Yes, feel free to have a go -- are you familiar enough with Theano and Lasagne? The idea would be to create a new UpscaleLayer class with a constructor `__init__(self, incoming, scale_factor, mode='repeat', axes='spatial', **kwargs)`. Oh, wait, did you just file #808 under another account or is that somebody else? Let's continue the discussion there.
